server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/change/StateChange.scala

```diff
@@ -46,62 +46,70 @@ trait TypeName {
       .getOrElse(Map())
 
   def asString(): String
-  def parse(string: String): TypeName
+  def parse(string: String): Either[IllegalArgumentException, TypeName]
```

Comment:
Or... `Option[TypeName]` as we go through all value, generating an error is not meaningful...